### PR TITLE
PeerAuthentication Merging for Ambient

### LINKED ISSUE #53884: PeerAuthentication with port specification not overriding global default in ambient mode
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

I have OpenTelemetry Operator running in sidecar mode with the following Peer Authentication set:

```
apiVersion: security.istio.io/v1
kind: PeerAuthentication
metadata:
  labels:
    argocd.argoproj.io/instance: apps-opentelemetry-operator
  name: opentelemetry-operator
  namespace: opentelemetry-operator-system
spec:
  portLevelMtls:
    '9443':
      mode: PERMISSIVE
  selector:
    matchLabels:
      control-plane: controller-manager
```
this allows workloads to hit its webhook without certificate errors.

If I switch the `opentelemetry-operator-system` namespace to ambient mode, I get errors like this:

```
ztunnel-c2kv8 istio-proxy 2024-11-13T17:36:10.380963Z   error   access  connection complete     src.addr=10.0.0.240:49872 dst.addr=10.0.21.243:9443 dst.service="opentelemetry-operator-webhook-service.opentelemetry-operator-system.svc.cluster.local" dst.workload="opentelemetry-operator-controller-manager-89f9575cd-q2nsp" dst.namespace="opentelemetry-operator-system" direction="inbound" bytes_sent=0 bytes_recv=0 duration="0ms" error="connection closed due to policy rejection: explicitly denied by: istio-system/istio_converted_static_strict"
```

and I have to modify the PeerAuthentication to the following to get it to work again:

```
apiVersion: security.istio.io/v1
kind: PeerAuthentication
metadata:
  labels:
    argocd.argoproj.io/instance: apps-opentelemetry-operator
  name: opentelemetry-operator
  namespace: opentelemetry-operator-system
spec:
  mtls:
    mode: PERMISSIVE
  selector:
    matchLabels:
      control-plane: controller-manager
```

I don't recall seeing this error in Istio 1.23 with ambient mode enabled.

### Version

```Text
❯ istioctl version
client version: 1.24.0
control plane version: 1.24.0
data plane version: 1.24.0 (13 proxies)
❯ kubectl version
Client Version: v1.30.4
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.30.4-eks-a737599
```

### Additional Information

[bug-report.tar.gz](https://github.com/user-attachments/files/17737020/bug-report.tar.gz)

### PR DESCRIPTION
**Pull request by keithmattix**
**Please provide a description of this PR:**
image

Fixes #53884

After the last episode of the Ambient PeerAuthentication chronicles, we discovered that attaching both the static STRICT policy (due to a mesh or namespace level policy) and an port-level permissive exception will still result in traffic being blocked because DENY policies are OR'd (only one has to match, and the STRICT policy always did). Thus, we have to merge PeerAuthentications at the control plane level in order to issue a single correct policy. Added some tests that will catch this moving forward along with the reverse. Reviewers, please recommend additional test cases that you see missing

P.S the docs on integration testing flags has annoyed me for a while, so I fixed it

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/ambient/authorization.go [hunk 1]
```diff
@@ -62,8 +62,20 @@ func (a *index) Policies(requested sets.Set[model.ConfigKey]) []model.WorkloadAu
...
+		if oldest == nil || pol.CreationTimestamp.Before(&oldest.CreationTimestamp) {
+			oldest = pol
+		}
+	}
+	return oldest
+}
+
 // convertedSelectorPeerAuthentications returns a list of keys corresponding to one or both of:
 // [static STRICT policy, port-level STRICT policy] based on the effective PeerAuthentication policy
+// TODO: If the workload peerauthn has a mode UNSET and a parent-level strict policy, merge and don't attach
```
Comment (howardjohn) [inline pilot/pkg/serviceregistry/kube/controller/ambient/authorization.go:77]: Why? it seems right as it is, sending the strict policy?

File: tests/integration/ambient/baseline_test.go [hunk 2]
```diff
@@ -787,13 +788,81 @@ spec:
   mtls:
     mode: STRICT
   portLevelMtls:
-    8080:
+    80:
```
Comment (howardjohn) [inline tests/integration/ambient/baseline_test.go:791]: Not sure this is right (nor was 8080). This should be workload port not service port, so 18080? https://github.com/istio/istio/blob/cbb6a2778189d08af99b3076b76fec31e245ea24/pkg/test/framework/components/echo/common/ports/ports.go#L24

**Review by howardjohn — COMMENTED** [review]
LGTM overall

File: tests/integration/ambient/baseline_test.go (hunk 2 above)
Comment (keithmattix) [reply tests/integration/ambient/baseline_test.go:791]: Ah, I knew there was something off here! Thanks

File: pilot/pkg/serviceregistry/kube/controller/ambient/authorization.go (hunk 1 above)
Comment (keithmattix) [reply pilot/pkg/serviceregistry/kube/controller/ambient/authorization.go:77]: Yeah this is incorrectly phrased, but also implemented (correctly). What I mean to say was "if the workload peerauthn has a mode UNSET with a port-level PERMISSIVE and a parent-level strict policy, merge and don't send the static strict pol". I just forgot to come back and clean this up

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
this might be the cause of some flakes like https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/54136/integ-ambient_istio/1863745610658615296

**Pull request merged**